stdlib/src/utils/string_slice.mojo

```diff
@@ -1108,6 +1291,150 @@ fn _to_string_list[
     return _to_string_list[len_fn, unsafe_ptr_fn](items)
 
 
+@always_inline
+fn _split[
+    T0: Stringlike, //,
+    has_maxsplit: Bool,
+    has_sep: Bool,
+    T1: Stringlike = StringLiteral,
+](src_str: T0, sep: Optional[T1], maxsplit: Int) -> List[String]:
+    @parameter
+    if has_sep:
+        return _to_string_list(
+            _split_impl[has_maxsplit](src_str, sep.value(), maxsplit)
+        )
+    else:
+        return _to_string_list(_split_impl[has_maxsplit](src_str, maxsplit))
+
+
+fn _split_sl[
```

Comment:
**Suggestion** Spell out `sl` here (presumably abbreviated as `StringSlice`) for clarity for future readers.